packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1391,22 +1391,60 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction hash associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param hash - The new hash value to be assigned.
+   */
+  updateTransactionHash(transactionId: string, hash: string) {
+    const transactionMeta = this.getTransaction(transactionId);
+    if (!transactionMeta) {
+      log(`Transaction with id ${transactionId} not found.`);
+      return;
+    }
+    transactionMeta.hash = hash;
+    this.updateTransaction(
+      transactionMeta,
+      'TransactionController:updateTransactionHash - Transaction hash updated',
+    );
+  }
+
+  /**
+   * Updates the transaction status associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param status - The new status value to be assigned.
+   */
+  updateTransactionStatus(transactionId: string, status: TransactionStatus) {
```

Comment:
As mentioned in the other comment, perhaps `updateCustodialTransactionStatus` and we could limit which status can be applied?

To prevent the state going backwards to `unapproved` for example.